### PR TITLE
Adding default Vector config, removing Datadog, and general cleanup

### PR DESCRIPTION
**Pull request by blarghmatey**
- Adding a default Vector config to collect host metrics
- Removed all assignment of Datadog to host targeting for install and configuration
- Removed references for edX and MongoDB in top files since we are no longer running those via Salt

### CODE REVIEW

File: pillar/top.sls [hunk 1]
```diff
@@ -223,54 +180,12 @@ base:
...
-    - vector.edx
-    - datadog.nginx-integration
-    - datadog.supervisord-integration
-  'P@roles:(edx|edx-worker)$and not G@roles:edx-analytics':
-    - match: compound
-    - vector.edx
-  'P@roles:(edx|edx-worker) and not G@edx_codename:tumbleweed':
-    - match: compound
-    - edx.ansible_vars.theme
   'G@roles:sandbox and P@environment:mitx(pro)?-qa':
```
Comment (shaidar) [inline pillar/top.sls:183]: Should this be removed as well?

Comment (blarghmatey) [reply pillar/top.sls:183]: Yeah, probably

File: salt/top.sls [hunk 2]
```diff
@@ -181,52 +163,11 @@ base:
     - match: compound
     - elasticsearch
     - elasticsearch.plugins
-  'G@roles:elasticsearch and P@environment:(micromasters|production-apps)':
-    - match: compound
-    - datadog
-    - datadog.plugins
   'roles:mongodb':
```
Comment (shaidar) [inline salt/top.sls:166]: Do we still need this?

File: salt/top.sls [hunk 3]
```diff
@@ -275,27 +216,13 @@ base:
     - match: compound
     - nginx
     - nginx.certificates
-  'P@roles:ocw-(cms|db|origin|mirror) and G@ocw-environment:production':
-    - datadog
   'roles:sandbox':
```
Comment (shaidar) [inline salt/top.sls:219]: We can probably remove this too

**Pull request merged**